File: data/src/main/java/com/arturogutierrez/openticator/domain/category/repository/CategoryRepositoryImpl.java
package com.arturogutierrez.openticator.domain.category.repository;

import com.arturogutierrez.openticator.domain.account.model.Account;
import com.arturogutierrez.openticator.domain.category.model.Category;
import com.arturogutierrez.openticator.storage.CategoryDiskDataStore;
import java.util.List;
import javax.inject.Inject;
import rx.Observable;

public class CategoryRepositoryImpl implements CategoryRepository {

  private final CategoryDataStore categoryDataStore;

  @Inject
  public CategoryRepositoryImpl(CategoryDiskDataStore categoryDiskDataStore) {
    this.categoryDataStore = categoryDiskDataStore;
  }

  @Override
  public Observable<Category> add(Category category) {
    return categoryDataStore.add(category);
  }

  @Override
  public Observable<Category> addAccount(Category category, Account account) {
    return categoryDataStore.addAccount(category, account);
  }

  @Override
  public Observable<List<Category>> getCategories() {
    return categoryDataStore.getCategories();
  }
}

File: data/src/main/kotlin/com/arturogutierrez/openticator/domain/category/repository/CategoryRepositoryImpl.kt
package com.arturogutierrez.openticator.domain.category.repository

import com.arturogutierrez.openticator.domain.account.model.Account
import com.arturogutierrez.openticator.domain.category.model.Category
import rx.Observable
import javax.inject.Inject

class CategoryRepositoryImpl @Inject constructor(val categoryDataStore: CategoryDataStore) : CategoryRepository {

  override fun add(category: Category): Observable<Category> {
    return categoryDataStore.add(category)
  }

  override fun addAccount(category: Category, account: Account): Observable<Category> {
    return categoryDataStore.addAccount(category, account)
  }

  override val categories: Observable<List<Category>>
    get() = categoryDataStore.categories
}
